File: src/main/java/go/scala/dao/CrudDAO.java
package go.scala.dao;

import java.util.List;

public interface CrudDAO<PK, M> {
	
	void persist(M model);
	
	M find(PK key);
	
	void remove(M model);
	
	M merge(M model);
	
	List<M> findAll(int offset, int maxResults);
	
}

File: src/main/scala/go/scala/dao/CrudDAO.scala
package go.scala.dao;

import java.util.List;

trait CrudDAO[PK, M] {
	
	def persist(model:M)
	
	def find(key:PK):M
	
	def remove(model:M)
	
	def merge(model:M):M
	
	def findAll(offset:Int, maxResults:Int):List[M]
	
}
